src/main/java/io/github/thebusybiscuit/slimefun4/core/accessibility/ConcreteAccessLevel.java

```diff
@@ -0,0 +1,79 @@
+package io.github.thebusybiscuit.slimefun4.core.accessibility;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Concrete access levels are defined to be constant and always present.
+ * These levels should be used a fallback for when fine-grained access
+ * processing is not needed or unavailable.
+ *
+ * @author md5sha256
+ */
+public class ConcreteAccessLevel implements AccessLevel {
+
+    /**
+     * Represents an access level where no privileges are allocated.
+     */
+    public static final ConcreteAccessLevel NONE = new ConcreteAccessLevel("&cNone");
+
+    /**
+     * Represents an access level where all privileges are allocated.
+     */
+    public static final ConcreteAccessLevel FULL = new ConcreteAccessLevel("&bFull");
+
+    private final String displayName;
+
+    private ConcreteAccessLevel(final String displayName) {
+        this.displayName = displayName;
+    }
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    @Nullable
+    @Override
+    public AccessLevel next() {
+        return this == NONE ? FULL : null;
+    }
+
+    @Nullable
+    @Override
+    public AccessLevel previous() {
+        return this == FULL ? NONE : null;
+    }
+
+    @Nonnull
+    @Override
+    public ConcreteAccessLevel toConcreteAccessLevel() {
+        return this;
+    }
```

Comment:
What exactly is the point of this?

Reply:
This was intended for handlers to be able handle the access level at a macro level with far less precision. For example, the android UI handler would (by default) only care if a person has full access to the android or not. However, this may change when more fine-grained access is needed say when specific actions of an android should be limited based off of an access level. An example of this might be "FarmerAndroid.ALLOW_REFUEL". I'm not entirely sure how to go about this but its intended so that the access levels have some coherence and structure. Handlers should not have to care about every single level, and only some, depending on precision. The concrete access level serves to be of the lowest precision so to speak.

Reply:
uuuh, hmmm. I think restricting access on a per-instruction basis is a bit out of scope for the intend behind this issue.
The original suggestion that was posted on discord only wanted a "friends list" for their androids so they could allow others to access the android's inventory or some very trusted individuals to even edit the source code of the android.
I think adding the ability for new integrations like a "Adhere to protection plugin" mode or such would be awesome.
But I'm not sure if this might be a bit too deep.

Reply:
Righty, well in this case I'm not sure how best to proceed for "future proofing". I'll leave the api-centric design for this as is sans the fine-grain access control for now.

Reply:
Honestly, as I've been going about implementing I *strongly* see that it would make sense for a "per-operation" access level to be in place alongside general ("concrete") access levels. For example, I'd imagine some players would friends to be able to execute scripts but not modify them. This would (therefore) require fine-grained access control.